### PR TITLE
@ #501 | should support id update with deprecation

### PR DESCRIPTION
**Pull request by hieptranquoc**
connect #501 
please check PR! thank you

### CODE REVIEW

**Comment by hieptranquoc** [comment]
@hoatle I've updated, please check this PR again!

**Review by datphan — DISMISSED** [review]

File: lib/teracy-dev/util.rb [hunk 1]
```diff
@@ -215,11 +215,20 @@ def self.override(originHash, sourceHash)
                 id_existing = false
                 originHash[key] ||= []
                 originHash[key].each do |val1|
+
+                  if val1['_id_deprecated'] != nil && val['_id'] == val1['_id_deprecated']
+                    @@logger.warn("The _id: #{val1['_id_deprecated']} is deprecated, use the _id: #{val1['_id']} instead")
```
Comment (hoatle) [inline lib/teracy-dev/util.rb]: @@logger.warn("The _id: '#{val1['_id_deprecated']}' is deprecated, use the _id: '#{val1['_id']}' instead for #{val}")

Comment (hoatle) [reply lib/teracy-dev/util.rb]: something wrong, I expect 1 time of the warning only, got 2 times:

```
$ vagrant status
[TeracyDev::Util][WARN]: The _id: '0' is deprecated, use the _id: 'kernel-core' instead for {"_id"=>"0", "location"=>{"git"=>{"remote"=>{"origin"=>"https://github.com/teracyhq-incubator/teracy-dev-core.git"}, "branch"=>"develop"}}, "require_version"=>">= 0.4.0-SNAPSHOT", "enabled"=>true}
/Users/hoatle/teracy-dev/lib/teracy-dev/util.rb:220:in `block (3 levels) in override'

[TeracyDev::Util][WARN]: The _id: '0' is deprecated, use the _id: 'kernel-core' instead for {"_id"=>"0", "location"=>{"git"=>{"remote"=>{"origin"=>"https://github.com/teracyhq-incubator/teracy-dev-core.git"}, "branch"=>"develop"}}, "require_version"=>">= 0.4.0-SNAPSHOT", "enabled"=>true}
/Users/hoatle/teracy-dev/lib/teracy-dev/util.rb:220:in `block (3 levels) in override'
```

**Review by hoatle — CHANGES_REQUESTED** [review]
@hieptranquoc pls check my comment

**Review by hoatle — APPROVED** [review]
LGTM, will do some minor clean up

**Pull request merged**